python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -253,6 +301,213 @@ def _process_data(user_pre_parser, user_parser,
                     task_info, dur)
     return return_dict
 
+def process_featless_ntype(node_confs: List[Dict],
+                           edge_confs: List[Dict],
+                           remap_id: bool,
+                           num_processes: int=1) -> Dict:
+    """ Handle the cases when users do not provide node files for
+        feature less node type(s),
+
+    The node config of a node type is defined as follows:
+    {
+        "node_id_col":  "<column name>",
+        "node_type":    "<node type>",
+        ...
+    }
+    The edge config of an edge type is defined as follows:
+    {
+        "source_id_col":    "<column name>",
+        "dest_id_col":      "<column name>",
+        "relation":         "<src type, relation type, dest type>",
+        ...
+    }
+
+    By checking the node_types from node configs and relations from
+    edge configs we can get the node type(s) that are missing from
+    node configs.
+
+    Parameters
+    ----------
+    node_confs: list of dicts
+        The configurations of node data.
+    edge_confs: list of dicts
+        The configurations of edge data.
+    remap_id: bool
+        Whether or not to remap node IDs
+    num_processes: int
+        The number of processes to process the input files.
+
+    Return
+    ------
+    node_id_map: dict
+        Node ID map.
+    """
+    node_types = set()
+    node_ids = {}
+    node_id_map = {}
+    for node_conf in node_confs:
+        # each iteration is to process a node type.
+        assert 'node_type' in node_conf, \
+                "'node_type' must be defined for a node type"
+        node_type = node_conf['node_type']
+        node_types.add(node_type)
+
+    for edge_conf in edge_confs:
+        assert 'relation' in edge_conf, \
+            "'relation' is not defined for an edge type."
+        edge_type = edge_conf['relation']
+        assert 'files' in edge_conf, \
+                "'files' is not defined for an edge type."
+        in_files = get_in_files(edge_conf['files'])
+        assert 'format' in edge_conf, \
+                "'format' is not defined for an edge type."
+
+        src_ntype = edge_type[0]
+        dst_ntype = edge_type[2]
+
+        multiprocessing = do_multiprocess_transform(edge_conf,
+                                                    feat_ops=None,
+                                                    label_ops=None,
+                                                    in_files=in_files)
+        # If it requires multiprocessing, we need to read data to memory.
+        read_file = parse_edge_file_format(edge_conf, in_mem=multiprocessing)
+        num_proc = num_processes if multiprocessing else 0
+
```

Comment:
The code block below have too many duplicated lines. I would recommend to refactor like:

```
parse_src = src_ntype in node_types
parse_dst = dst_ntype in node_types

user_parser = partial(parse_edge_nid_data, parse_src=parse_src, parse_dst=parse_dst ...)

if parse_src:
   ...
```